src/vos/vos_common.c

```diff
@@ -310,9 +310,12 @@ vos_tx_end(struct vos_container *cont, struct dtx_handle *dth_in,
 			goto cancel;
 		}
 
-		dru = &dth->dth_rsrvds[dth->dth_rsrvd_cnt++];
+		dru          = &dth->dth_rsrvds[dth->dth_rsrvd_cnt];
 		dru->dru_scm = *rsrvd_scmp;
-		*rsrvd_scmp = NULL;
+		if (*rsrvd_scmp != NULL) {
+			*rsrvd_scmp = NULL;
+			dth->dth_rsrvd_cnt++;
+		}
```

Comment:
On second thought, I think original code won't cause issue since the NULL rsrvd_act will be ignored in umem_cancel(). I need to investigate the issue further.